### PR TITLE
Async Thread

### PR DESCRIPTION
**Pull request by leameow**
@aviau for review!

Lots of stuff crammed in this PR, but the main goal is to have the first wrapper to call purple from async thread.

### CODE REVIEW

File: src/icq/protocol.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        platform: "web",
+        create_account: "1",
+        client: "icq",
+        r: &random_id(),
+    };
+    let login_response = client::login_with_phone_number(&login_with_phone_number_body)
+        .await
+        .map_err(Error::ApiError)?;
+    log::info!("Login response: {:?}", login_response);
+    Ok(RegisteredAccountInfo {})
```
Comment (aviau) [inline src/icq/protocol.rs:58]: Rien for now? Ou juste pas implémenté?

File: src/icq/system.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                Err(error) => {
+                    log::error!("Failed to receive message: {:?}", error);
+                    break;
+                }
+            };
+            log::info!("Message: {:?}", purple_message);
+            match purple_message {
+                PurpleMessage::Login(account_info) => self.login(account_info).await,
+            }
+            self.ping().await
```
Comment (aviau) [inline src/icq/system.rs:56]: Tu ping le serveur nonstop?

**Comment by aviau** [comment]
LGTM

File: src/icq/protocol.rs (hunk 1 above)
Comment (leameow) [reply src/icq/protocol.rs:58]: wip!

File: src/icq/system.rs (hunk 2 above)
Comment (leameow) [reply src/icq/system.rs:56]: Pour test les messages, removed in following PR

**Pull request merged**